### PR TITLE
Add `calc.random(seed, min, max)` function

### LINKED ISSUE #984: Pseudo-random number generation
### Description

`calc.random(seed, min, max)` would be a new function under the Calculate category, that takes three optional numbers and, using the first number as a seed, returns a random number between the remaining two numbers.

## To discuss

- Default `min` and `max` values:
    - **Suggested answer**: 0 and 100
- The way to get the random numbers
Rust's standard library doesn't contain a RNG. So we need to pull in a dependency to get those numbers.
    - **Suggested answer**: [`fastrand`](https://crates.io/crates/fastrand)
- Should this function be in `calc` or, should it really be in `math`?

As the function returns an integer, both `min` and `max` would get truncated to be integers.
Seed would be settable.

---

I have a working implementation for this, I'm just opening this issue for discussion before the PR.

### Use Case

Everyone that would need a random number generated in compile-time.

### PR DESCRIPTION
**Pull request by blyxyas**
Adds a `random()` function to the `calc` category. This function takes 3 optional arguments:

- `seed`: The RNG seed (default: `0x4d595df4d0f33173`)
- `min`: The minimum number in range (default: 0)
- `max`: The maximum number in range (default: 100)

Note that this RNG is deterministic, as we considered that, the same input should **always** return the same output.

Fixes #984

### CODE REVIEW

File: library/src/compute/calc.rs [hunk 1]
```diff
@@ -962,6 +964,40 @@ pub fn quo(
...
+/// ## Example
+/// ```example
+/// #assert(calc.random(seed: 1) == calc.random(seed: 1))
+/// #assert(calc.random(seed: 1) != calc.random(seed: 2))
+/// ```
+///
+/// Display: Random
+/// Category: Calculate
+/// Returns: integer
+pub fn random(
```
Comment (HarmoGlace) [inline library/src/compute/calc.rs:981]: Could you create a random float function too?

Comment (blyxyas) [reply library/src/compute/calc.rs:982]: Added

File: library/src/compute/calc.rs [hunk 2]
```diff
@@ -962,6 +965,67 @@ pub fn quo(
...
+/// #assert(calc.rand_int(seed: 1) != calc.rand_int(seed: 2))
+/// ```
+///
+/// Display: Random integer
+/// Category: Calculate
+/// Returns: integer
+pub fn rand_int(
+    /// The seed, optional
+    #[named]
+    #[default(0x4d595df4d0f33173)]
```
Comment (szdytom) [inline library/src/compute/calc.rs:985]: it is a bad idea to use this magic number 4 times in the code, maybe u should define a constant value like `default_seed`

Comment (blyxyas) [reply library/src/compute/calc.rs:985]: Is this for a change in behaviour, or just for readability?

Comment (szdytom) [reply library/src/compute/calc.rs]: just for readability :)

**Comment by laurmaedje** [comment]
Marked as draft until consensus about API is reached in #984.

**Comment by laurmaedje** [comment]
As there hasn't really been any progress on this and the design isn't clear yet, I'm going to close this. Still, thanks for your work!

**Pull request closed without merging**